modules/prbot/src/main/scala/ru/bugzmanov/prcheck/checks/JavaPmdExecutor.scala

```diff
@@ -0,0 +1,81 @@
+package ru.bugzmanov.prcheck.checks
+
+import java.io.BufferedInputStream
+import javax.annotation.concurrent.NotThreadSafe
+
+import net.sourceforge.pmd._
+import net.sourceforge.pmd.lang.Language
+import net.sourceforge.pmd.lang.java.JavaLanguageModule
+import net.sourceforge.pmd.util.datasource.DataSource
+
+import scala.util.{Failure, Success, Try}
+
+@NotThreadSafe
+class JavaPmdExecutor private (config: PMDConfiguration) {
+  val BlacklistedRules = Set("LawOfDemeter", "LongVariable", "AbstractNaming", "JUnitTestsShouldIncludeAssert", "JUnitSpelling")
+  val ruleSetFactory = new RuleSetFactory
+  val processor = new SourceCodeProcessor(config)
+  val rules = RulesetsFactoryUtils.getRuleSets(config.getRuleSets, ruleSetFactory)
+  val ctx = new RuleContext
+
+  def analyze(source: DataSource): Try[Iterable[ViolationIssue]] = {
```

Comment:
Yeah, I know that [it's wiser to default to Vector, rather than List when choosing default Seq implementation](http://stackoverflow.com/a/6934116/298389). Likely I will change it to Vector.